docs/command-line-interface.md

```diff
@@ -0,0 +1,1085 @@
+
+```
+export PATH="~/polaris:$PATH"
+```
+
+Alternatively, you can run the CLI by providing a path to it, such as with the following invocation:
+
+```
+~/polaris principals list
+```
+
+## Commands
+
+Each of the commands `catalogs`, `principals`, `principal-roles`, `catalog-roles`, and `privileges` is used to manage a different type of entity within Polaris.
+
+To find details on the options that can be provided to a particular command or subcommand ad-hoc, you may wish to use the `--help` flag. For example:
+
+```
+polaris catalogs --help
+polaris principals create --help
+```
+
+### catalogs
+
+The `catalogs` command is used to create, discover, and otherwise manage catalogs within Polaris.
+
+`catalogs` supports the following subcommands:
+
+1. create
+2. delete
+3. get
+4. list
+5. update
+
+#### create
+
+The `create` subcommand is used to create a catalog.
+
+```
+input: polaris catalogs create --help
+options:
+  create
+    Named arguments:
+      --type  The type of catalog to create in [INTERNAL, EXTERNAL]. INTERNAL by default.
+      --storage-type  (Required) The type of storage to use for the catalog
+      --default-base-location  (Required) Default base location of the catalog
+      --allowed-location  An allowed location for files tracked by the catalog. Multiple locations can be provided by specifying this option more than once.
+      --role-arn  (Required for S3) A role ARN to use when connecting to S3
+      --external-id  (Only for S3) The external ID to use when connecting to S3
+      --tenant-id  (Required for Azure) A tenant ID to use when connecting to Azure Storage
+      --multi-tenant-app-name  (Only for Azure) The app name to use when connecting to Azure Storage
+      --consent-url  (Only for Azure) A consent URL granting permissions for the Azure Storage location
+      --service-account  (Only for GCS) The service account to use when connecting to GCS
+      --remote-url  (For external catalogs) The remote URL to use
+      --property  A key/value pair such as: tag=value. Multiple can be provided by specifying this option more than once
+    Positional arguments:
+      catalog
+```
+
+##### Examples
+
+```
+polaris catalogs create \
+  --storage-type s3 \
+  --default-base-location s3://example-bucket/my_data \
+  --role-arn ${ROLE_ARN} \
+  my_catalog
+
+polaris catalogs create \
+  --storage-type s3 \
+  --default-base-location s3://example-bucket/my_other_data \
+  --allowed-location s3://example-bucket/second_location \
+  --allowed-location s3://other-bucket/third_location \
+  --role-arn ${ROLE_ARN} \
+  my_other_catalog
+```
+
+#### delete
+
+The `delete` subcommand is used to delete a catalog.
+
+```
+input: polaris catalogs delete --help
+options:
+  delete
+    Positional arguments:
+      catalog
+```
+
+##### Examples
+
+```
+polaris catalogs delete some_catalog
+
+polaris catalogs delete another_catalog
```

Comment:
nit: Why do we need `polaris catalogs delete another_catalog` as an example? Seems the same as` polaris catalogs delete some_catalog`.

Reply:
Oh. seems we are trying to show two examples?